lib/string-checker.js

```diff
@@ -130,7 +130,12 @@ StringChecker.prototype = {
 
         this._configuredRules.forEach(function(rule) {
             errors.setCurrentRule(rule.getOptionName());
-            rule.check(file, errors);
+
+            try {
```

Comment:
Anyway, code in this branch is faster than in master (about 5%). `try-catch` is almost made like @petkaantonov recommends. Still don't see a problem here ;-(

An article is not a real benchmark. Optimizations (that try-catch prevents) requires a lot of other things to happen. 

Personally I don't like try-catchs but I don't see another way here to catch an exception and to increase stability (and friendliness).